src/main/java/uk/gov/hmcts/reform/fees/client/deserialization/DeserializationUtils.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.hmcts.reform.fees.client.deserialization;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+class DeserializationUtils {
+    private DeserializationUtils() {
+        // no op
+    }
+
+    static <FIELD> void extractNode(
+            Consumer<FIELD> setter,
+            Function<JsonNode, FIELD> nodeToValue,
+            JsonNode node,
+            String... children
+    ) {
+        Optional<JsonNode> focus = Optional.ofNullable(node);
+        for (String child : children) {
+            focus = focus.map(n -> n.get(child));
```

Comment:
my head is 💥. I am a bit confused by this for loop - you're overriding focus at each iteration without accumulating anything (ie. with a reduce), doesn't this mean that you're just going to have focus set as the last child when the for loop ends?

Reply:
Yes, this is safely (because `Optional`) navigating the JSON tree to find the node we want to extract the data from.